packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift

```diff
@@ -156,37 +155,74 @@ struct FlutterSerializedModel: Model, JSONValueHolder {
         }
     }
 
-    private func generateSerializedData(modelSchema: ModelSchema) -> [String: Any]{
-        
+    private static func getModelSchema(flutterModelRegistration: FlutterModels, modelName: String) throws -> ModelSchema {
+        do {
+            return try FlutterDataStoreRequestUtils.getModelSchema(modelSchemas: flutterModelRegistration.modelSchemas, modelName: modelName)
+        } catch let error {
+            print("model \(modelName) is not registered.")
+            throw error
+        }
+    }
+
+    private static func generateSerializedJsonData(values: [String: JSONValue], flutterModelRegistration: FlutterModels, modelName: String) throws -> [String: Any] {
+        let modelSchema = try getModelSchema(flutterModelRegistration: flutterModelRegistration, modelName: modelName)
         var result = [String: Any]()
-                
+
+        for (key, value) in values {
+            let field = modelSchema.field(withName: key)
+            if case .object(let deserializedValue) = value {
+                // If a field that has many models
+                if (deserializedValue["associatedField"] != nil && deserializedValue["associatedId"] != nil) {
+                    result[key] = Array<Any>();
+                }
+                // If a field that has one or belongs to a model
+                else if case .string(let modelId) = deserializedValue["id"],
+                        case .model(let nextModelName) = field!.type {
+                    result[key] = [
+                        "id": modelId,
+                        "modelName": nextModelName,
+                        "serializedData": try generateSerializedJsonData(values: deserializedValue, flutterModelRegistration: flutterModelRegistration, modelName: nextModelName)
+                    ]
+                }
+            } else if case .string(let deserializedValue) = value {
```

Comment:
*Question:* Can these `else if` statements be combined into one `else` ? The other cases are mostly handled the same way.

Reply:
Yes absolutely :) I was pretty much following the examples in the codebase, it looks like a convention that to explicitly list all possible types and the way to handle the value decoding/casting respectively. I can merge them into one `else`

Reply:
I'm going to keep it as is, as the if else flow also includes value parsing using `JSONValue` method, which cannot be combined. Thanks!